src/blob/core.rs

```diff
@@ -282,19 +295,16 @@ where
         }))
     }
 
-    pub(crate) async fn mark_all_as_deleted(&mut self, key: &K) -> Result<Option<u64>> {
-        if self.index.get_any(key).await?.is_some() {
-            let on_disk = self.index.on_disk();
-            if on_disk {
-                self.load_index().await?;
-            }
-            let record = Record::deleted(key)?;
-            self.write(record).await?;
-            let res = self.index.mark_all_as_deleted(key)?;
-            Ok(res)
-        } else {
-            Ok(None)
+    pub(crate) async fn mark_all_as_deleted(&mut self, key: &K) -> Result<u64> {
```

Comment:
I think, this function should be renamed. It is now adds a deletion record, so possible name can be `put_deletion_record`.
Also, it would be nice to keep the old `mark_all_as_deleted` function, which first checks for existence in the index and then calls the new one.